meeshkan/serve/mock/rest.py

```diff
@@ -30,15 +31,18 @@ def add(self, url):
     def spew(
         self, request: Request, specs: Sequence[OpenAPISpecification]
     ) -> Sequence[OpenAPISpecification]:
+        if len(self._endpoints) == 0:
+            return specs
+
         req = HttpExchangeWriter.to_dict(request)
         cs = {spec.source: convert_from_openapi(spec.api) for spec in specs}
         for endpoint in self._endpoints:
             res = requests.post(endpoint, json={"request": req, "schemas": cs})
             cs = res.json()
-        return [
-            OpenAPISpecification(convert_to_openapi(api), source)
-            for source, api in cs.items()
-        ]
-
 
-rest_middleware_manager = RestMiddlewareManager()
+        out: List[OpenAPISpecification] = []
+        for name, dict_spec in cs.items():
+            spec = convert_to_openapi(dict_spec)
+            self._mock_data_store.add_mock(name, spec)
+            out.append(OpenAPISpecification(spec, name))
```

Comment:
I think this might be easier to read when split into two separate actions, like
```python
specs_result = [OpenAPISpecification(convert_to_openapi(dict_spec), name) for name, dict_spec in cs.items()]
for oai in specs_result:
  self._mock_data_store.add_mock(oai.name, oai.spec)
```